tests/wallet/rpc/test_wallet_rpc.py

```diff
@@ -571,12 +575,17 @@ async def test_cat_endpoints(wallet_rpc_environment: WalletRpcTestEnvironment):
     assert should_be_none is None
     assert name == next(iter(DEFAULT_CATS.items()))[1]["name"]
 
-    # TODO: Investigate why farming only one block here makes it flaky
-    for _ in range(5):
+    # make sure spend is in mempool before farming tx block
+    await time_out_assert(5, check_mempool_spend_count, True, full_node_api, 1)
+    for i in range(5):
+        if check_mempool_spend_count(full_node_api, 0):
```

Comment:
this is still a hack, im not sure yet why we sometimes farm empty blocks and not include the spend even though it is in the mempool

but this is a better hack then what we had

Reply:
Thanks for digging into all of this.